internal/hostname/ec2/ec2.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package ec2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/hostname/cachedfetch"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/hostname/httputils"
+)
+
+// declare these as vars not const to ease testing
+var (
+	metadataURL     = "http://169.254.169.254/latest/meta-data"
```

Comment:
Interesting, so I'm just copying the agent behavior here (which maybe knows that this is possible?) According to the docs here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html AWS recommends using `latest` since it refers not to the API version but to the metadata version? I'm not familiar with compatibility guarantees and a quick search in AWS doesn't seem to reveal anything, any insight?

Reply:
TIL! Should be fine if it's recommended by AWS and used by the agent. Thank you for checking it out!